Topic updates for SDK 2.0.0 with breaking changes, migration steps, release notes (#691)
RevealBi/Documentation: docs/web/upgrade-guide-v2.0.0.md

## Overview of Breaking Changes

- **jQuery and Day.js removed** — the SDK no longer depends on jQuery or Day.js.

> - **NPM delivery** — the client SDK is now delivered as an npm package; legacy script-tag delivery is no longer the recommended approach.

Reviewer: ```suggestion
- **NPM delivery** — the client SDK is now delivered as an npm package. Legacy script-tag delivery is no longer the recommended approach.
```

- **Renamed and removed APIs** 

>     - `DateFilter` - _removed_ deprecated property from `RevealView`, `RVDashboard` and `ExportOptionsBase`

Reviewer: ```suggestion
    - `DateFilter` - _removed_ deprecated property from `RevealView`, `RVDashboard`, `RVDateDashboardFilter`, `IExportOptions` (and all classes implementing it), and `RevealSettings`.
```

    - `Reveal.Sdk.Dashboard.ToJsonStringAsync` - _renamed_ to `ToJsonString`.

> - **Deprecated types** — `RVDashboardThumbnailView` has been deprecated, in favor of `RVThumbnail`.

Reviewer: ```suggestion
- **Deprecated types** — `RVDashboardThumbnailView` has been _deprecated_, in favor of `RVThumbnail`.
```

## Step-by-Step Upgrade

### 1. Remove jQuery

The Reveal SDK no longer requires jQuery. Remove the jQuery script tag that was loaded for the SDK:

```html
<!-- Remove this line -->
<script src="https://cdn.jsdelivr.net/npm/jquery@3.6.0/dist/jquery.min.js"></script>
```

:::info

> If your own application code depends on jQuery you can keep it — the Reveal SDK simply no longer requires it.

Reviewer: ```suggestion
If your own application code depends on jQuery, you can keep it. It is simply no longer required by the Reveal SDK.
```

</Tabs>

:::tip Still need script tags?
The SDK distribution zip is still available for non-bundler setups — jQuery and Day.js are no longer needed:

```html
<script src="./assets/reveal/reveal-sdk.js"></script>
```
:::

:::info Full installation guides
For a deeper dive into all client SDK installation options — including CDN with ES modules, npm with Angular, and npm with React — see the [Installation Guide](installation/installation.md).
:::

### 3. Update server SDK packages

<Tabs groupId="code" queryString>
  <TabItem value="aspnet" label="ASP.NET" default>

Update the `Reveal.Sdk.*` NuGet packages to version **2.0.0** or later.

```xml
<PackageReference Include="Reveal.Sdk.AspNetCore" Version="2.0.0" />
```

  </TabItem>
  <TabItem value="java" label="Java">

Update your Maven/Gradle dependency to version **2.0.0** or later.

```xml
<dependency>
    <groupId>com.infragistics.reveal.sdk</groupId>
    <artifactId>reveal-sdk</artifactId>
    <version>2.0.0</version>
</dependency>
```

  </TabItem>
  <TabItem value="node" label="Node.js">

```bash
npm install reveal-sdk-node@2.0.0
```

  </TabItem>
</Tabs>

### 4. Update API usage

#### `DateFilter` → `filters` + `RVDateRule`

> The deprecated `DateFilter` property has been removed. Use the `filters` collection instead.  DateFilter was eliminated from `RevealView`, `RVDashboard`, `RVDateDashboardFilter`, `IExportOptions`, `RevealSettings`, `ExportOptionsBase` and child classes.

Reviewer: ```suggestion
The deprecated `DateFilter` property has been removed. Use the `filters` collection instead.  DateFilter was eliminated from `RevealView`, `RVDashboard`, `RVDateDashboardFilter`, `IExportOptions` (and all classes implementing it), and `RevealSettings`.
```

<Tabs groupId="api-datefilter">
  <TabItem value="before" label="1.x">

```javascript
var myRule = new Reveal.RVDateRule(Reveal.RVPeriodRelation.Last, 3, Reveal.RVPeriodType.Month);
dashboard.dateFilter = new Reveal.RVDateDashboardFilter(myRule);
```

  </TabItem>
  <TabItem value="after" label="2.0">

```javascript
var myRule = new Reveal.RVDateRule(Reveal.RVPeriodRelation.Last, 3, Reveal.RVPeriodType.Month);
var myDateFilter = dashboard.filters.findByTitle("My Date Filter");
myDateFilter.rule = myRule;
```

  </TabItem>
</Tabs>

#### `RVDashboardThumbnailView` → `RVThumbnail`

<Tabs groupId="api-thumbnail">
  <TabItem value="before" label="1.x">

```javascript
var thumbnailView = new Reveal.RevealDashboardThumbnailView("#thumbnail");
Reveal.RevealUtility.getDashboardInfo("Sales", function (info) {
  thumbnailView.dashboardInfo = info.info;
});
```

  </TabItem>
  <TabItem value="after" label="2.0">

```javascript
Reveal.RVThumbnail.fromDashboard("#thumbnail", "Sales");
```

  </TabItem>
</Tabs>

> The new `RVThumbnail` API also supports runtime theme changes.

Reviewer: ```suggestion
The new `RVThumbnail` API also responds to runtime theme changes.
```

## Removed APIs

| API | Replacement |
|---|---|
| `DateFilter` property | `filters` collection |
| `RVDashboardThumbnailView` | `RVThumbnail` |
| `Reveal.Sdk.Dashboard.ToJsonStringAsync` | `ToJsonString` |

> | Legacy chart types (previously deprecated) | Use current [Chart Types](/web/chart-types) |

Reviewer: ```suggestion
| Legacy chart types (previously deprecated) | Use current [chart types](/web/chart-types) |
```

| Legacy Java engine | Java SDK (Spring Boot) |

## Summary Checklist

- [ ] Remove jQuery `<script>` tag
- [ ] Remove Quill.js and Spectrum.js references if still present
- [ ] Switch client SDK to npm package (or remove jQuery from script-tag setup)
- [ ] Update server SDK packages to 2.0.0
- [ ] Replace uses of `DateFilter` property to use `Filters` list instead.

> - [ ] Replace `RVDashboardThumbnailView` with `RVThumbnail`

Reviewer: ```suggestion
- [ ] (recommended) Replace `RVDashboardThumbnailView` with `RVThumbnail`
```

## Overview of Breaking Changes

- **jQuery removed** — the SDK no longer depends on jQuery.
- **NPM delivery** — the client SDK is now delivered as an npm package; legacy script-tag delivery is no longer the recommended approach.
- **Legacy Java engine & WPF backend removed** — these have been completely removed.
- **Legacy chart types removed** — previously deprecated chart types are no longer available.

> - **Renamed and removed APIs** — `DateFilter`, filter property names, `RVDashboardThumbnailView`, and `ToJsonStringAsync` have been renamed or removed.

Reviewer: What is it referring to by "filter property names" here?

Author: I have changed the whole content here to:
- **Renamed and removed APIs** 
    - `DateFilter` - _removed_ deprecated property from `RevealView`, `RVDashboard` and `ExportOptionsBase`
    - `ToJsonStringAsync` - _renamed_ to `ToJsonString`.

> - **Renamed and removed APIs** — `DateFilter`, filter property names, `RVDashboardThumbnailView`, and `ToJsonStringAsync` have been renamed or removed.

Reviewer: IIRC RVDashboardThumbnailView is only marked as deprecated for the time being, with no other changes.
```suggestion
- **Renamed and removed APIs** — `DateFilter`, filter property names, and `ToJsonStringAsync` have been renamed or removed.
- **Deprecated types** — `RVDashboardThumbnailView` has been deprecated, in favor of RVThumbnail.
```

```

  </TabItem>
  <TabItem value="after" label="2.0 (npm)">

```bash
npm install reveal-sdk dayjs
```

```typescript
import "reveal-sdk";
```

  </TabItem>
</Tabs>

:::tip Still need script tags?
The SDK distribution zip is still available for non-bundler setups — but jQuery is no longer needed:

```html
<script src="https://unpkg.com/dayjs@1.8.21/dayjs.min.js"></script>
<script src="./assets/reveal/infragistics.reveal.js"></script>
```
:::

### 3. Update server SDK packages

<Tabs groupId="code" queryString>
  <TabItem value="aspnet" label="ASP.NET" default>

Update the `Reveal.Sdk.*` NuGet packages to version **2.0.0** or later.

```xml
<PackageReference Include="Reveal.Sdk.AspNetCore" Version="2.0.0" />
```

  </TabItem>
  <TabItem value="java" label="Java">

Update your Maven/Gradle dependency to version **2.0.0** or later.

```xml
<dependency>
    <groupId>com.infragistics.reveal.sdk</groupId>
    <artifactId>reveal-sdk</artifactId>
    <version>2.0.0</version>
</dependency>
```

  </TabItem>
  <TabItem value="node" label="Node.js">

```bash
npm install reveal-sdk-node@2.0.0
```

  </TabItem>
</Tabs>

### 4. Update API usage

> #### `DateFilter` → `filters` + `RVDateRule`

Reviewer: DateFilter was removed from many places.
It would probably be a good idea to mention them all (RevealView, RVDashboard RVDateDashboardFilter, IExportOptions, RevealSettings come to mind).
https://github.com/Infragistics-BusinessTools/Reveal/pull/3196/changes

Author: Addressed

The deprecated `DateFilter` property has been removed. Use the `filters` collection instead.

<Tabs groupId="api-datefilter">
  <TabItem value="before" label="1.x">

```javascript
visualization.dateFilter = new RVDateDashboardFilter();
```

  </TabItem>
  <TabItem value="after" label="2.0">

```javascript
const dateRule = new RVDateRule();
visualization.filters = [dateRule];
```

  </TabItem>
</Tabs>

> #### Filter property renames

Reviewer: These are not public API.

Author: Removed now

| Old Name (1.x) | New Name (2.0) |
|---|---|
| `NumberOfItemsInGrid` | `FilterCount` |
| `FilterRangeText` | `FilterSelectionText` |
| `UpdateFilterRangeText` | `UpdateFilterSelectionText` |

Update all references — the old names are no longer available.

#### `RVDashboardThumbnailView` → `RVThumbnail`

<Tabs groupId="api-thumbnail">
  <TabItem value="before" label="1.x">

```javascript
const thumbnailView = new RVDashboardThumbnailView();
```

  </TabItem>
  <TabItem value="after" label="2.0">

```javascript
const thumbnail = new RVThumbnail();
```

  </TabItem>
</Tabs>

The new `RVThumbnail` API also supports runtime theme changes.

#### `ToJsonStringAsync` → `ToJsonString`

<Tabs groupId="api-json">
  <TabItem value="before" label="1.x">

```csharp
var json = await dashboard.ToJsonStringAsync();
```

  </TabItem>
  <TabItem value="after" label="2.0">

```csharp
var json = dashboard.ToJsonString();
```

  </TabItem>
</Tabs>

## Removed APIs

| API | Replacement |
|---|---|

> | `DateFilter` property | `filters` collection with `RVDateRule` |

Reviewer: I'd remove the "with RVDateRule" remark, as that's not always the case, for example when opening preexisting dashboards. RVDateRule is only used when trying to set a new rule, but I think we're better off leaving it out.

Author: Addressed

| `RVDashboardThumbnailView` | `RVThumbnail` |
| `ToJsonStringAsync` | `ToJsonString` |

> | `NumberOfItemsInGrid` | `FilterCount` |

Reviewer: NumberOfItemsInGrid, FilterRangeText and UpdateFilterRangeText are not public API.

Author: Addressed

Reviewer: Issued suggestion removing these.

| `FilterRangeText` | `FilterSelectionText` |
| `UpdateFilterRangeText` | `UpdateFilterSelectionText` |
| Legacy chart types (previously deprecated) | Use current [Chart Types](/web/chart-types) |
| Legacy Java engine | Java SDK (Spring Boot) |
| WPF backend | ASP.NET, Node.js, or Java server SDKs |

## Summary Checklist

- [ ] Remove jQuery `<script>` tag
- [ ] Remove Quill.js and Spectrum.js references if still present
- [ ] Switch client SDK to npm package (or remove jQuery from script-tag setup)
- [ ] Update server SDK packages to 2.0.0
- [ ] Replace `DateFilter` usage with `filters` and `RVDateRule`

> - [ ] Rename `NumberOfItemsInGrid` → `FilterCount`

Reviewer: Another mention of these three.

- [ ] Rename `FilterRangeText` → `FilterSelectionText`
- [ ] Rename `UpdateFilterRangeText` → `UpdateFilterSelectionText`

> - [ ] Replace `RVDashboardThumbnailView` with `RVThumbnail`

Reviewer: Recommended, but not mandatory for the time being.

## Overview of Breaking Changes

- **jQuery and Day.js removed** — the SDK no longer depends on jQuery or Day.js.
- **NPM delivery** — the client SDK is now delivered as an npm package; legacy script-tag delivery is no longer the recommended approach.

> - **Legacy Java engine & WPF backend removed** — these have been completely removed.

Reviewer 2: 1. WPF is irrelevant here, since all this documentation is centered in the web platforms. There was already a separate announcement regarding the deprecation of the WPF SDK, back in March.
2. Regarding java, this overview should just mention that 2.0 comes with a LOT of breaking changes. Customers don't care about 'legacy engine', so I wouldn't mention that.

Author: Addressed

- **Legacy chart types removed** — previously deprecated chart types are no longer available.
- **Renamed and removed APIs** 
    - `DateFilter` - _removed_ deprecated property from `RevealView`, `RVDashboard` and `ExportOptionsBase`

>     - `ToJsonStringAsync` - _renamed_ to `ToJsonString`.

Reviewer: We need to specify in what class this method was renamed.

```suggestion
    - `Reveal.Sdk.Dashboard.ToJsonStringAsync` - _renamed_ to `ToJsonString`.
```

Author: addressed

    - `NumberOfItemsInGrid` - _renamed_ to `FilterCount`.
    - `FilterRangeText` - _renamed_ to `FilterSelectionText`.

>     - `UpdateFilterRangeText` - _renamed_ to `UpdateFilterSelectionText`.

Reviewer: ```suggestion
```
These need to go.

Author: addressed

- **Deprecated types** — `RVDashboardThumbnailView` has been deprecated, in favor of `RVThumbnail`.

## Step-by-Step Upgrade

### 1. Remove jQuery

The Reveal SDK no longer requires jQuery. Remove the jQuery script tag that was loaded for the SDK:

```html
<!-- Remove this line -->
<script src="https://cdn.jsdelivr.net/npm/jquery@3.6.0/dist/jquery.min.js"></script>
```

:::info
If your own application code depends on jQuery you can keep it — the Reveal SDK simply no longer requires it.
:::

> Also remove **Quill.js** and **Spectrum.js** if they are still present from older versions:

Reviewer 2: and Dayjs

Author: addressed

```html
<!-- Remove these if present -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/spectrum/1.8.0/spectrum.min.css" rel="stylesheet" type="text/css" >
<script src="https://cdnjs.cloudflare.com/ajax/libs/spectrum/1.8.0/spectrum.min.js"></script>
<link href="https://cdn.quilljs.com/1.3.6/quill.snow.css" rel="stylesheet" type="text/css" >
<script src="https://cdn.quilljs.com/1.3.6/quill.min.js"></script>
```

### 2. Switch client SDK to NPM

> Replace the legacy script-tag installation with the npm package.

Reviewer 2: This section is the old way of things

Author: @MLopezIG this is how its structured, it shows how it was for 1.x version and now with 2.0

<img width="1920" height="911" alt="image" src="https://github.com/user-attachments/assets/fcba76c9-ab05-4deb-afdc-c2d79843411d" />
<img width="1920" height="911" alt="image" src="https://github.com/user-attachments/assets/f4b360c2-f1fb-4ef2-8c99-80f63129927e" />

<script src="https://unpkg.com/dayjs@1.8.21/dayjs.min.js"></script>
<script src="https://dl.revealbi.io/reveal/libs/1.8.4/infragistics.reveal.js"></script>
```

  </TabItem>
  <TabItem value="after" label="2.0 (npm)">

```bash
npm install reveal-sdk
```

```typescript
import "reveal-sdk";
```

  </TabItem>
</Tabs>

:::tip Still need script tags?
The SDK distribution zip is still available for non-bundler setups — jQuery and Day.js are no longer needed:

```html
<script src="./assets/reveal/reveal-sdk.js"></script>
```
:::

:::info Full installation guides
For a deeper dive into all client SDK installation options — including CDN with ES modules, npm with Angular, and npm with React — see the [Installation Guide](installation/installation.md).
:::

### 3. Update server SDK packages

<Tabs groupId="code" queryString>
  <TabItem value="aspnet" label="ASP.NET" default>

Update the `Reveal.Sdk.*` NuGet packages to version **2.0.0** or later.

```xml
<PackageReference Include="Reveal.Sdk.AspNetCore" Version="2.0.0" />
```

  </TabItem>
  <TabItem value="java" label="Java">

Update your Maven/Gradle dependency to version **2.0.0** or later.

```xml

> <dependency>

Reviewer 2: One of the big changes in java is that the group and artifact have changed:

```
        <dependency>
            <groupId>io.revealbi</groupId>
            <artifactId>reveal-sdk-servlet</artifactId>
            <version>2.0.0</version>
        </dependency>
```

but this is just one of many changes. I suggest using one of the new samples and copy the relevant parts here. ... but I see, we don't have yet! I guess we will once Pablo finiishes with the samples work...

Author: @PVillarmarzo  FYI

    <groupId>com.infragistics.reveal.sdk</groupId>
    <artifactId>reveal-sdk</artifactId>
    <version>2.0.0</version>
</dependency>
```

  </TabItem>
  <TabItem value="node" label="Node.js">

```bash
npm install reveal-sdk-node@2.0.0
```

  </TabItem>
</Tabs>

### 4. Update API usage

#### `DateFilter` → `filters` + `RVDateRule`

The deprecated `DateFilter` property has been removed. Use the `filters` collection instead.  DateFilter was eliminated from `RevealView`, `RVDashboard`, `RVDateDashboardFilter`, `IExportOptions`, `RevealSettings`, `ExportOptionsBase` and child classes.

<Tabs groupId="api-datefilter">
  <TabItem value="before" label="1.x">

```javascript

> visualization.dateFilter = new RVDateDashboardFilter();

Reviewer: Code is not right.
```suggestion
var myRule = new Reveal.RVDateRule(Reveal.RVPeriodRelation.Last, 3, Reveal.RVPeriodType.Month);
dashboard.dateFilter = new Reveal.RVDateDashboardFilter(myRule);
```

Author: addressed

```

  </TabItem>
  <TabItem value="after" label="2.0">

```javascript
const dateRule = new RVDateRule();

> visualization.filters = [dateRule];

Reviewer: Code is not right.
```suggestion
var myRule = new Reveal.RVDateRule(Reveal.RVPeriodRelation.Last, 3, Reveal.RVPeriodType.Month);
var myDateFilter = dashboard.filters.findByTitle("My Date Filter");
myDateFilter.rule = myRule;
```

Author: addressed

```

  </TabItem>
</Tabs>

#### `RVDashboardThumbnailView` → `RVThumbnail`

<Tabs groupId="api-thumbnail">
  <TabItem value="before" label="1.x">

```javascript

> const thumbnailView = new RVDashboardThumbnailView();

Reviewer: ```suggestion
var thumbnailView = new RevealApi.RevealDashboardThumbnailView("#thumbnail");
RevealApi.RevealUtility.getDashboardInfo("Sales", function (info) {
    thumbnailView.dashboardInfo = info.info;
  });
```

Author: Done

```

  </TabItem>
  <TabItem value="after" label="2.0">

```javascript

> const thumbnail = new RVThumbnail();

Reviewer: ```suggestion
RevealApi.RVThumbnail.fromDashboard("#thumbnail", "Sales");
```

Author: addressed

```

  </TabItem>
</Tabs>

The new `RVThumbnail` API also supports runtime theme changes.

#### Renamed filter properties

The following properties have been renamed:

| 1.x | 2.0 |
|---|---|
| `NumberOfItemsInGrid` | `FilterCount` |
| `FilterRangeText` | `FilterSelectionText` |

> | `UpdateFilterRangeText` | `UpdateFilterSelectionText` |

Reviewer: These are not public API.
```suggestion
```

Author: removed

> #### `ToJsonStringAsync` → `ToJsonString`

Reviewer: I'm not sure I'd provide sample code for a renamed method.

Author: removed

<Tabs groupId="api-json">
  <TabItem value="before" label="1.x">

```csharp
var json = await dashboard.ToJsonStringAsync();
```

  </TabItem>
  <TabItem value="after" label="2.0">

```csharp
var json = dashboard.ToJsonString();
```

  </TabItem>
</Tabs>

## Removed APIs

| API | Replacement |
|---|---|
| `DateFilter` property | `filters` collection |
| `RVDashboardThumbnailView` | `RVThumbnail` |
| `ToJsonStringAsync` | `ToJsonString` |
| `NumberOfItemsInGrid` | `FilterCount` |
| `FilterRangeText` | `FilterSelectionText` |

> | `UpdateFilterRangeText` | `UpdateFilterSelectionText` |

Reviewer: These are not public API.
```suggestion
```

Author: Removed both from upgrade guide and from release notes

| Legacy chart types (previously deprecated) | Use current [Chart Types](/web/chart-types) |
| Legacy Java engine | Java SDK (Spring Boot) |
| WPF backend | ASP.NET, Node.js, or Java server SDKs |

## Summary Checklist

- [ ] Remove jQuery `<script>` tag
- [ ] Remove Quill.js and Spectrum.js references if still present
- [ ] Switch client SDK to npm package (or remove jQuery from script-tag setup)
- [ ] Update server SDK packages to 2.0.0

> - [ ] Replace `DateFilter` usage with `filters` and `RVDateRule`

Reviewer: ```suggestion
- [ ] Replace uses of `DateFilter` property to use `Filters` list instead
```

Author: Done

- [ ] Replace `RVDashboardThumbnailView` with `RVThumbnail`

> - [ ] Replace `ToJsonStringAsync` with `ToJsonString`

Reviewer: ```suggestion
- [ ] Replace `Reveal.Sdk.Dashboard.ToJsonStringAsync` with `ToJsonString`
```

Author: addressed

> - [ ] Rename `NumberOfItemsInGrid` to `FilterCount`, `FilterRangeText` to `FilterSelectionText`, and `UpdateFilterRangeText` to `UpdateFilterSelectionText`

Reviewer: ```suggestion
```

Author: Gone
